core/src/main/java/org/elasticsearch/index/query/HasParentQueryParser.java

```diff
@@ -30,9 +30,10 @@
 public class HasParentQueryParser implements QueryParser<HasParentQueryBuilder>  {
 
     private static final HasParentQueryBuilder PROTOTYPE = new HasParentQueryBuilder("", EmptyQueryBuilder.PROTOTYPE);
-    private static final ParseField QUERY_FIELD = new ParseField("query", "filter");
-    private static final ParseField SCORE_FIELD = new ParseField("score_mode").withAllDeprecated("score");
-    private static final ParseField TYPE_FIELD = new ParseField("parent_type", "type");
+    public static final ParseField QUERY_FIELD = new ParseField("query", "filter");
+    public static final ParseField SCORE_MODE_FIELD = new ParseField("score_mode").withAllDeprecated("score");
```

Comment:
For reference after chatting f2f: Before switching to ParseFields we did output "score" as fieldname in the builder which if continuing to use this syntax going forward we would not. Simplifying to use new ParseField("score", "score_mode") instead.